### ISSUE TITLE
does not fail when commits message don't follow conventional commits

### ISSUE THREAD

**Issue by electriquo** [semantic-release/commit-analyzer#278]
This is a snippet from my configuration

```js
plugins: [
  ["@semantic-release/commit-analyzer", {
    preset: "conventionalcommits",
  }],
  ["@semantic-release/release-notes-generator", {
    preset: "conventionalcommits",
   }],
  ...
```
and and logs indicate that it was loaded

```txt
semantic-release:plugins options for @semantic-release/commit-analyzer/analyzeCommits: { preset: 'conventionalcommits' }
[semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: foo
semantic-release:commit-analyzer Analyzing with default rules +0ms
[@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[@semantic-release/commit-analyzer] › ℹ  Analysis of 1 commits complete: no release
› ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
```

If it matters have installed [conventional-changelog-conventionalcommits](https://github.com/conventional-changelog/conventional-changelog/releases/tag/conventional-changelog-conventionalcommits-v4.6.1).

I would expect the plugin to fail semantic release when the git commit message history contains a commit that does not follow conventional commit.

In the documentation it is written

> For presets that expects a configuration object, such as conventionalcommits, the presetConfig option must be set.

maybe this is the reason that the plugin does not fail ([what should be the `presetConfig`?](https://github.com/semantic-release/commit-analyzer/issues/263)).

What am I missing? Is this a bug?

Thank you

**Comment by gr2m** [comment]
> I would expect the plugin to fail semantic release when the git commit message history contains a commit that does not follow conventional commit.

This is by design. We don't enforce that all commits have to follow the specified conventions, it would be too late for it anyway, the commits exist in the releasing branch at this point. It's something you might want to check for before you push commits into a branch that `semantic-release` runs on, some folks like to use https://github.com/zeke/semantic-pull-requests for that purpose.

**Comment by electriquo** [comment]
@gr2m Thank you. Could you share information regarding the `presetConfig`?

**Comment by dargmuesli** [comment]
[Seems like](https://github.com/semantic-release/commit-analyzer/blob/ab3ee9f93b02f3d7d288323ee2fc89c59d64290d/test/load-parser-config.test.js#L93) it is enough to set `presetConfig: {}`.
